tests/phpunit/SCQ_CompoundQueryApiTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+
+use SMW\Tests\Utils\MwApiFactory;
+
+/**
+ * @covers SCQCompoundQueryApi
+ * @ingroup SemanticCompoundQueries
+ *
+ * @author Peter Grassberger < petertheone@gmail.com >
+ */
+
+class SCQCompoundQueryApiTest extends \PHPUnit_Framework_TestCase
+{
+	/**
+	 * @var MwApiFactory
+	 */
+	private $apiFactory;
+
+	protected function setUp() {
+		parent::setUp();
+		$this->apiFactory = new MwApiFactory();
+	}
+
+	public function testCanConstruct() {
+		$instance = new SCQCompoundQueryApi(
+			$this->apiFactory->newApiMain( array( 'query' => 'Foo' ) ),
+			'compoundquery'
+		);
+		$this->assertInstanceOf(
+			'SCQCompoundQueryApi',
+			$instance
+		);
+	}
+
+	/**
+	 * @param array $query
+	 * @param array $expected
+	 *
+	 * @dataProvider sampleQueryProvider
+	 */
+	public function testExecute( array $query, array $expected ) {
+		$results = $this->apiFactory->doApiRequest( array(
+			'action' => 'compoundquery',
+			'query' => implode( '|', $query )
+		) );
+		$this->assertInternalType( 'array', $results );
+		// If their is no printrequests array we expect an error array
+		if ( isset( $results['query']['printrequests'] ) ) {
+			$this->assertEquals( $expected, $results['query']['printrequests'] );
+		} else {
+			$this->assertArrayHasKey( 'error', $results );
+		}
+	}
+
+	public function sampleQueryProvider() {
+		// #0 Standard query
```

Comment:
It's hard to figure out what is actually tested. Apparently "standard query", and then two things which apparently are not standard. I'm not a fan of data provider usage like this (you would not need the if-else in `testExecute` if you did it without), though if you go down that path, you can add descriptive messages as keys for the argument arrays:

$provider['Foo and bar should result in baz'] = array( /* ... */ );